src/app/shared/components/course-card/course-card.component.ts

```diff
@@ -1,16 +1,26 @@
-import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  Output,
+} from '@angular/core';
+import { Courses } from 'src/app/models/course.interface';
 
 @Component({
   selector: 'course-card',
   templateUrl: './course-card.component.html',
   styleUrls: ['./course-card.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class CourseCardComponent implements OnInit {
+export class CourseCardComponent {
+  @Input()
+  public course: Courses;
 
-  constructor() { }
-
-  ngOnInit(): void {
-  }
+  @Output()
+  public emitDeleteCourse: EventEmitter<number> = new EventEmitter<number>();
```

Comment:
use like this `emitDeleteCourse = new EventEmitter<number>();`
The type is gonna be inferred when you immediately assign the value. Also, you can skip the `public` keyword, all props are `public` by default)

Reply:
done